README.md

```diff
@@ -1,2 +1,167 @@
+- [Goals [or Motivating Use Cases, or Scenarios]](#goals-or-motivating-use-cases-or-scenarios)
+- [Non-goals](#non-goals)
+- [User research](#user-research)
+- [[API 1]](#api-1)
+- [[API 2]](#api-2)
+- [Key scenarios](#key-scenarios)
+  - [Scenario 1](#scenario-1)
+  - [Scenario 2](#scenario-2)
+- [Detailed design discussion](#detailed-design-discussion)
+  - [[Tricky design choice #1]](#tricky-design-choice-1)
+  - [[Tricky design choice 2]](#tricky-design-choice-2)
+- [Considered alternatives](#considered-alternatives)
+  - [[Alternative 1]](#alternative-1)
+  - [[Alternative 2]](#alternative-2)
+- [Stakeholder Feedback / Opposition](#stakeholder-feedback--opposition)
+- [References & acknowledgements](#references--acknowledgements)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Introduction
+
+TODO: Fill this in using https://tag.w3.org/explainers/ as a reference.
+
+[The "executive summary" or "abstract".
+Explain in a few sentences what the goals of the project are,
+and a brief overview of how the solution works.
+This should be no more than 1-2 paragraphs.]
+
+## Goals [or Motivating Use Cases, or Scenarios]
+
+[What is the **end-user need** which this project aims to address?]
+
+## Non-goals
+
+[If there are "adjacent" goals which may appear to be in scope but aren't,
+enumerate them here. This section may be fleshed out as your design progresses and you encounter necessary technical and other trade-offs.]
+
+## User research
+
+[If any user research has been conducted to inform the design choices presented
+discuss the process and findings.
+We strongly encourage that API designers consider conducting user research to
+verify that their designs meet user needs and iterate on them,
+though we understand this is not always feasible.]
+
+<!-- In your initial explainer, you shouldn't be attached or appear attached to any of the potential
+solutions you describe below this. -->
+
+## [Potential Solution 1]
+
+[For each related element of the proposed solution - be it an additional JS method, a new object, a new element, a new concept etc., create a section which briefly describes it.]
+
+```js
+// Provide example code - not IDL - demonstrating the design of the feature.
+
+// If this API can be used on its own to address a user need,
+// link it back to one of the scenarios in the goals section.
+
+// If you need to show how to get the feature set up
+// (initialized, or using permissions, etc.), include that too.
+```
+
+[Where necessary, provide links to longer explanations of the relevant pre-existing concepts and API.
+If there is no suitable external documentation, you might like to provide supplementary information as an appendix in this document, and provide an internal link where appropriate.]
+
+[If this is already specced, link to the relevant section of the spec.]
+
+[If spec work is in progress, link to the PR or draft of the spec.]
+
+## [Potential Solution 2]
```

Comment:
This is possible, but relatively unlikely (that a second+ solution would be documented, right?  Perhaps we should remove this, but above say "there may be more than one potential solution section".

Reply:
Sounds good.